frontend/src/components/utils/Footer/ContactModal.tsx

```diff
@@ -0,0 +1,719 @@
+      name: name,
+      address: address,
+      photos,
+      userId: user?.uid,
+    };
+  };
+
+  const updateApartmentName = (event: React.ChangeEvent<HTMLInputElement>) => {
+    apartmentDispatch({ type: 'updateApartmentName', name: event.target.value });
+  };
+
+  const updateApartmentAddress = (event: React.ChangeEvent<HTMLInputElement>) => {
+    apartmentDispatch({ type: 'updateApartmentAddress', address: event.target.value });
+  };
+
+  const updateApartmentPhotos = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { files } = event.target;
+    if (!files) return;
+    const availablePhotos = PHOTOS_LIMIT - apartmentForm.localPhotos.length;
+    if (availablePhotos <= 0) {
+      console.log(`Max file limit of ${PHOTOS_LIMIT} exceeded`);
+      return;
+    }
+
+    const newFiles = [...files].slice(0, availablePhotos);
+    const bigPhoto = newFiles.find((newFiles) => newFiles.size > PHOTO_MAX_MB * Math.pow(1024, 2));
+    if (bigPhoto) {
+      console.log(`File ${bigPhoto.name} exceeds max size of ${PHOTO_MAX_MB}`);
+      return;
+    }
+    apartmentDispatch({
+      type: 'updatePhotos',
+      photos: [...apartmentForm.localPhotos, ...newFiles],
+    });
+  };
+
+  const removePhoto = (index: number) => {
+    const newPhotos = apartmentForm.localPhotos.filter((_, photoIndex) => index !== photoIndex);
+    apartmentDispatch({ type: 'updatePhotos', photos: newPhotos });
+  };
+
+  //  Question Modal Constants/Functions
+  const questionFormDataToReview = async ({
+    name,
+    email,
+    msg,
+  }: QuestionFormData): Promise<QuestionForm> => {
+    return {
+      date: new Date(),
+      name: name,
+      email: email,
+      msg: msg,
+      userId: user?.uid,
+    };
+  };
+
+  const updateQuestionName = (event: React.ChangeEvent<HTMLInputElement>) => {
+    questionDispatch({ type: 'updateQuestionName', name: event.target.value });
+  };
+
+  const updateQuestionEmail = (event: React.ChangeEvent<HTMLInputElement>) => {
+    questionDispatch({ type: 'updateQuestionEmail', email: event.target.value });
+  };
+
+  const updateQuestionMsg = (event: React.ChangeEvent<HTMLInputElement>) => {
+    questionDispatch({ type: 'updateQuestionMsg', msg: event.target.value });
+  };
+
+  // Toast
+  const showToast = (setState: (value: React.SetStateAction<boolean>) => void) => {
+    setState(true);
+    setTimeout(() => {
+      setState(false);
+    }, TOAST_TIME);
+  };
+  const showConfirmationToast = (type: string) => {
+    setConfirmationType(type);
+    showToast(setShowConfirmation);
+  };
+
+  //  onSubmit functions for each modal
+  const onApartmentSubmit = async () => {
+    try {
+      setSending(true);
+      const token = await user!.getIdToken(true);
+      const data = await apartmentFormDataToReview(apartmentForm);
+      if (data.name === '' || includesProfanity(data.name) || includesProfanity(data.address)) {
+        data.name === '' ? setEmptyNameError(true) : setEmptyNameError(false);
```

Comment:
Minor suggestion, this could be simplified to: setEmptyNameError(data.name === ''). Although it doesn't need changing since your current code is good for readability.